libfoundation/src/foundation-number.cpp

```diff
@@ -207,6 +306,73 @@ bool MCNumberParseUnicodeChars(const unichar_t *p_chars, uindex_t p_char_count,
 
 ////////////////////////////////////////////////////////////////////////////////
 
+bool MCNumberFormat(MCNumberRef self, MCStringRef& r_string)
+{
+    switch(__MCNumberGetRepType(self))
+    {
+        case __kMCNumberRepSignedInteger:
+            return MCStringFormat(r_string, "%d", self -> integer);
+        case __kMCNumberRepUnsignedInteger:
+            return MCStringFormat(r_string, "%u", self -> unsigned_integer);
+        case __kMCNumberRepDouble:
+            return MCStringFormat(r_string, "%lf", self -> real);
+        default:
+            MCUnreachable();
+            break;
+    }
+
+    return false;
+}
+
+bool MCNumberTryToParse(MCStringRef p_string, MCNumberRef& r_number)
```

Comment:
Can you add a variant of this function that accepts an offset to begin parsing from, and that returns the number of characters consumed rather than throwing an error if the whole string isn't used?